File: app/src/main/java/edu/uark/cartapp/commands/converters/ByteToUUIDConverterCommand.java
package edu.uark.cartapp.commands.converters;

import java.nio.ByteBuffer;
import java.util.UUID;

import edu.uark.cartapp.commands.interfaces.ResultInterface;

/* ==== APP ByteToUUIDConverterCommand.java ====*/
public class ByteToUUIDConverterCommand implements ResultInterface<UUID> {
	private byte[] valueToConvert;

	@Override
	public UUID execute() {
		ByteBuffer byteBuffer = ByteBuffer.wrap(this.valueToConvert);
		return new UUID(byteBuffer.getLong(), byteBuffer.getLong());
	}

	public byte[] getValueToConvert() {
		return this.valueToConvert;
	}

	public ByteToUUIDConverterCommand setValueToConvert(byte[] valueToConvert) {
		this.valueToConvert = valueToConvert;
		return this;
	}
}

File: app/src/main/java/edu/uark/cartapp/commands/converters/ByteToUUIDConverterCommand.kt
package edu.uark.cartapp.commands.converters

import java.nio.ByteBuffer
import java.util.UUID

import edu.uark.cartapp.commands.interfaces.ResultInterface

/* ==== APP ByteToUUIDConverterCommand.java ====*/
class ByteToUUIDConverterCommand : ResultInterface<UUID> {
	private var valueToConvert: ByteArray? = null

	override fun execute(): UUID {
		val byteBuffer = ByteBuffer.wrap(this.valueToConvert)
		return UUID(byteBuffer.long, byteBuffer.long)
	}

	fun getValueToConvert(): ByteArray? {
		return this.valueToConvert
	}

	fun setValueToConvert(valueToConvert: ByteArray): ByteToUUIDConverterCommand {
		this.valueToConvert = valueToConvert
		return this
	}
}